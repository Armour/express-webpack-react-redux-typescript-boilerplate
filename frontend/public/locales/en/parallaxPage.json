{
  "title": "Parallax Page",
  "description": "Parallax is an effect where the background content or image in this case, is moved at a different speed than the foreground content while scrolling."
}
