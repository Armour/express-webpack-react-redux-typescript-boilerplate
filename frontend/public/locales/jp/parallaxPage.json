{
  "title": "視差ページ",
  "description": "Parallax(パララックス)は、この場合のバックグラウンドコンテンツまたは画像がスクロール中にフォアグラウンドコンテンツとは異なる速度で移動される場合の効果です。"
}
