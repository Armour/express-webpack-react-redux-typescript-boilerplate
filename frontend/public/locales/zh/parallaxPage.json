{
  "title": "视差页面",
  "description": "Parallax(视差) 是指一种背景内容或图像在滚动时以与前景内容不同的速度移动的效果。"
}
