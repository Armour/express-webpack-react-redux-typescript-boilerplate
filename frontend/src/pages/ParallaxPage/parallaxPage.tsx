import React, { Fragment } from 'react';
import { Translation } from 'react-i18next';

import PrismCodes, { PARALLAX_CODE } from './components/PrismCodes';

const styles = require('./parallaxPage.scss');

export class ParallaxPage extends React.Component {
  public componentDidMount() {
    const elems = document.querySelectorAll('.parallax');
    M.Parallax.init(elems);
  }

  public render() {
    return (
      <Translation ns='parallaxPage'>
        {(t) => (
          <Fragment>
            <div className='white'>
              <h1 className='page-title'>{t('title')}</h1>
            </div>
            <div className='parallax-container'>
              <div className='parallax'>
                <img className='parallax-img' src={require('./assets/images/parallax1.jpg')} alt='parallax-img' />
              </div>
            </div>
            <div className='section white'>
              <div className='row container'>
                <h2 className={styles['parallax-header']}>Parallax</h2>
                <p className='grey-text text-darken-3'>{t('description')}</p>
              </div>
              <div className='row container'>
                <h4 className='light'>Parallax Demo Code</h4>
                <PrismCodes language='language-tsx'>
                  {PARALLAX_CODE}
                </PrismCodes>
              </div>
            </div>
            <div className='parallax-container'>
              <div className='parallax'>
                <img className='parallax-img' src={require('./assets/images/parallax2.jpg')} alt='parallax-img' />
              </div>
            </div>
          </Fragment>
        )}
      </Translation>
    );
  }
}

export default ParallaxPage;
