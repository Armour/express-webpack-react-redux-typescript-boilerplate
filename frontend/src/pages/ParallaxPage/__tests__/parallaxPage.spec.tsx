import 'materialize-css';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import { ParallaxPage } from '../parallaxPage';

jest.mock('../components/PrismCodes', () => 'PrismCodes');

describe('ParallaxPage', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <ParallaxPage />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
