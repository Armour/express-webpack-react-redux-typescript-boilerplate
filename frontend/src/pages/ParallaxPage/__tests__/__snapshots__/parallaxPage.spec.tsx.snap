// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParallaxPage should renders correctly 1`] = `
Array [
  <div
    className="white"
  >
    <h1
      className="page-title"
    >
      title
    </h1>
  </div>,
  <div
    className="parallax-container"
  >
    <div
      className="parallax"
    >
      <img
        alt="parallax-img"
        className="parallax-img"
        src={
          Object {
            "default": "test-file-stub",
          }
        }
      />
    </div>
  </div>,
  <div
    className="section white"
  >
    <div
      className="row container"
    >
      <h2
        className="parallax-header"
      >
        Parallax
      </h2>
      <p
        className="grey-text text-darken-3"
      >
        description
      </p>
    </div>
    <div
      className="row container"
    >
      <h4
        className="light"
      >
        Parallax Demo Code
      </h4>
      <PrismCodes
        language="language-tsx"
      />
    </div>
  </div>,
  <div
    className="parallax-container"
  >
    <div
      className="parallax"
    >
      <img
        alt="parallax-img"
        className="parallax-img"
        src={
          Object {
            "default": "test-file-stub",
          }
        }
      />
    </div>
  </div>,
]
`;
